error: no member named 'value' in 'pp::field

full error message

```

In file included from /tmp/tmp.ylbpg80qQ3/bug.cpp:1:
/tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:406:72: error: no member named 'value' in 'pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>'
                        n += skipper<typename F::coder>::encode_skip(f.value());
                                                                     ~ ^
/usr/bin/../lib/gcc/x86_64-redhat-linux/10.2.0/../../../../include/c++/10.2.0/type_traits:2506:7: note: in instantiation of function template specialization 'pp::skipper<pp::message_coder<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>>::encode_skip(const pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>> &)::(anonymous class)::operator()<pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>' requested here
      std::declval<_Fn>()(std::declval<_Args>()...)
      ^
/usr/bin/../lib/gcc/x86_64-redhat-linux/10.2.0/../../../../include/c++/10.2.0/type_traits:2517:24: note: while substituting deduced template arguments into function template '_S_test' [with _Fn = (lambda at /tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:394:26), _Args = (no value)]
      typedef decltype(_S_test<_Functor, _ArgTypes...>(0)) type;
                       ^
/usr/bin/../lib/gcc/x86_64-redhat-linux/10.2.0/../../../../include/c++/10.2.0/type_traits:2523:14: note: in instantiation of template class 'std::__result_of_impl<false, false, (lambda at /tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:394:26), pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>' requested here
    : public __result_of_impl<
             ^
/usr/bin/../lib/gcc/x86_64-redhat-linux/10.2.0/../../../../include/c++/10.2.0/type_traits:2845:29: note: in instantiation of template class 'std::__invoke_result<(lambda at /tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:394:26), pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>' requested here
                               __void_t<typename _Result::type>>
                                                 ^
/usr/bin/../lib/gcc/x86_64-redhat-linux/10.2.0/../../../../include/c++/10.2.0/type_traits:2962:7: note: during template argument deduction for class template partial specialization '__is_invocable_impl<_Result, _Ret, true>' [with _Result = std::__invoke_result<(lambda at /tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:394:26), pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>, _Ret = void]
    : __is_invocable_impl<__invoke_result<_Fn, _ArgTypes...>, void>::type
      ^
/usr/bin/../lib/gcc/x86_64-redhat-linux/10.2.0/../../../../include/c++/10.2.0/type_traits:2962:7: note: (skipping 19 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
/tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:201:41: note: while checking the satisfaction of concept 'invocable<(lambda at /tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:354:26), pp::field<{{115, 116, 117, 100, 101, 110, 116, 115, 0}}, 1, pp::embedded_message_coder<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>>>' requested here
        template <typename F> requires (std::invocable<F, T> && ...)
                                        ^~~~~~~~~~~~~~~~~~~~
/tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:201:46: note: while substituting template arguments into constraint expression here
        template <typename F> requires (std::invocable<F, T> && ...)
                                             ^~~~~~~~~~~~~~~
/tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:354:17: note: while checking constraint satisfaction for template 'for_each<(lambda at /tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:354:26)>' required here
            msg.for_each([&b]<field_c F> (const F& f) {
                ^~~~~~~~
/tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:354:17: note: in instantiation of function template specialization 'pp::message<pp::field<{{115, 116, 117, 100, 101, 110, 116, 115, 0}}, 1, pp::embedded_message_coder<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>>>::for_each<(lambda at /tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:354:26)>' requested here
/tmp/tmp.ylbpg80qQ3/bug.cpp:18:25: note: in instantiation of member function 'pp::message_coder<pp::message<pp::field<{{115, 116, 117, 100, 101, 110, 116, 115, 0}}, 1, pp::embedded_message_coder<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>>>>::encode' requested here
  message_coder<Class>::encode(c, buffer);
                        ^
In file included from /tmp/tmp.ylbpg80qQ3/bug.cpp:1:
/tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:203:14: error: no matching function for call to object of type '(lambda at /tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:394:26)'
            (std::forward<F>(f)(static_cast<const T&>(*this)), ...);
             ^~~~~~~~~~~~~~~~~~
/tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:394:17: note: in instantiation of function template specialization 'pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>::for_each<(lambda at /tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:394:26)>' requested here
            msg.for_each([&n]<field_c F> (const F& f) {
                ^
/tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:425:49: note: in instantiation of member function 'pp::skipper<pp::message_coder<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>>::encode_skip' requested here
            auto n = skipper<message_coder<T>>::encode_skip(msg);
                                                ^
/tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:366:39: note: in instantiation of member function 'pp::embedded_message_coder<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>::encode' requested here
                        b = F::coder::encode(i, b);
                                      ^
/usr/bin/../lib/gcc/x86_64-redhat-linux/10.2.0/../../../../include/c++/10.2.0/type_traits:2506:7: note: in instantiation of function template specialization 'pp::message_coder<pp::message<pp::field<{{115, 116, 117, 100, 101, 110, 116, 115, 0}}, 1, pp::embedded_message_coder<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>>>>::encode(const pp::message<pp::field<{{115, 116, 117, 100, 101, 110, 116, 115, 0}}, 1, pp::embedded_message_coder<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>>> &, pp::bytes)::(anonymous class)::operator()<pp::field<{{115, 116, 117, 100, 101, 110, 116, 115, 0}}, 1, pp::embedded_message_coder<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>>>' requested here
      std::declval<_Fn>()(std::declval<_Args>()...)
      ^
/usr/bin/../lib/gcc/x86_64-redhat-linux/10.2.0/../../../../include/c++/10.2.0/type_traits:2517:24: note: while substituting deduced template arguments into function template '_S_test' [with _Fn = (lambda at /tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:354:26), _Args = (no value)]
      typedef decltype(_S_test<_Functor, _ArgTypes...>(0)) type;
                       ^
/usr/bin/../lib/gcc/x86_64-redhat-linux/10.2.0/../../../../include/c++/10.2.0/type_traits:2523:14: note: (skipping 7 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
    : public __result_of_impl<
             ^
/tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:201:41: note: while checking the satisfaction of concept 'invocable<(lambda at /tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:354:26), pp::field<{{115, 116, 117, 100, 101, 110, 116, 115, 0}}, 1, pp::embedded_message_coder<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>>>' requested here
        template <typename F> requires (std::invocable<F, T> && ...)
                                        ^~~~~~~~~~~~~~~~~~~~
/tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:201:46: note: while substituting template arguments into constraint expression here
        template <typename F> requires (std::invocable<F, T> && ...)
                                             ^~~~~~~~~~~~~~~
/tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:354:17: note: while checking constraint satisfaction for template 'for_each<(lambda at /tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:354:26)>' required here
            msg.for_each([&b]<field_c F> (const F& f) {
                ^~~~~~~~
/tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:354:17: note: in instantiation of function template specialization 'pp::message<pp::field<{{115, 116, 117, 100, 101, 110, 116, 115, 0}}, 1, pp::embedded_message_coder<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>>>::for_each<(lambda at /tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:354:26)>' requested here
/tmp/tmp.ylbpg80qQ3/bug.cpp:18:25: note: in instantiation of member function 'pp::message_coder<pp::message<pp::field<{{115, 116, 117, 100, 101, 110, 116, 115, 0}}, 1, pp::embedded_message_coder<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{105, 100, 0}}, 1, pp::varint_coder<unsigned int>, pp::singular, std::vector<unsigned int>>, pp::field<{{110, 97, 109, 101, 0}}, 3, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>, pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>>>>>>::encode' requested here
  message_coder<Class>::encode(c, buffer);
                        ^
/tmp/tmp.ylbpg80qQ3/thirdparty/protopuf/include/protopuf/message.h:394:26: note: candidate template ignored: substitution failure [with F = pp::field<{{98, 111, 111, 107, 115, 0}}, 5, pp::embedded_message_coder<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>, pp::repeated, std::vector<pp::message<pp::field<{{110, 97, 109, 101, 0}}, 1, pp::array_coder<pp::integer_coder<char>, std::basic_string<char>>, pp::singular, std::vector<std::basic_string<char>>>>>>]
            msg.for_each([&n]<field_c F> (const F& f) {
                         ^
2 errors generated.
gmake[3]: *** [CMakeFiles/bug.dir/bug.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/bug.dir/all] Error 2
gmake[1]: *** [CMakeFiles/bug.dir/rule] Error 2
gmake: *** [bug] Error 2

```


test code

```cpp
#include "protopuf/message.h"
using namespace pp;
int main() {
  using Book = message<
      string_field<"name", 1>
      >;
  using Student = message<
      uint32_field<"id", 1>,
      string_field<"name", 3>,
      message_field<"books", 5, Book, repeated>
      >;
  using Class = message<
      message_field<"students", 1, Student, repeated>
      >;

  std::array<std::byte, 64> buffer{};
  Class c;
  message_coder<Class>::encode(c, buffer);
  return 0;
}
```


env
clang 13 with libstdc++ (gcc 10.2)

--- a/include/protopuf/message.h
+++ b/include/protopuf/message.h
@@ -403,7 +403,7 @@ namespace pp {
                 } else {
                     for(const auto &i : f) {
                         n += skipper<varint_coder<uint<4>>>::encode_skip(F::key);
-                        n += skipper<typename F::coder>::encode_skip(f.value());
+                        n += skipper<typename F::coder>::encode_skip(i);
                     }
                 }
             });

--- a/test/message.cpp
+++ b/test/message.cpp
@@ -97,6 +97,24 @@ GTEST_TEST(message_coder, encode) {
                                           0x05_b, 0x68_b, 0x65_b, 0x6c_b, 0x6c_b, 0x6f_b}));
     }
 
+    {
+        using Book = message<
+            string_field<"name", 1>
+        >;
+        using Student = message<
+            uint32_field<"id", 1>,
+            string_field<"name", 3>,
+            message_field<"books", 5, Book, repeated>
+        >;
+        using Class = message<
+            message_field<"students", 1, Student, repeated>
+        >;
+
+        std::array<std::byte, 64> buffer{};
+        Class c;
+        message_coder<Class>::encode(c, buffer);
+    }
+
 }
 
 GTEST_TEST(message_coder, decode) {

